server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -163,7 +165,9 @@ public synchronized void updateSegments(final SegmentInfos infos) throws IOExcep
      * @throws IOException - When there is an IO error committing the SegmentInfos.
      */
     private void commitSegmentInfos(SegmentInfos infos) throws IOException {
-        store.commitSegmentInfos(infos, localCheckpointTracker.getMaxSeqNo(), localCheckpointTracker.getProcessedCheckpoint());
+        if (shouldCommit) {
```

Comment:
nit: `shouldCommit == true` 
This applies to other places as well.

Reply:
I think convention is to only do this in the false case?